### ISSUE TITLE
Can't load library libAaptNativeWrapper64.dylib on macos

### ISSUE THREAD

**Issue by d3xter-lab** [apk-spectrum/apk-scanner#2]
When AaptNativeScanner loads libAaptNativeWrapper64.dylib, it failed. It looks like an architecture issue.
All libraries and executables in 'darwin' directory seems to be x86_64. Are there any plans to support arm64(M1 chip)?

# Error
```
Exception in thread "Thread-14" java.lang.UnsatisfiedLinkError: Can't load library: /xxxx/apk-scanner/target/tool/darwin/libAaptNativeWrapper64.dylib
at java.base/java.lang.ClassLoader.loadLibrary(ClassLoader.java:2633)
at java.base/java.lang.Runtime.load0(Runtime.java:768)
at java.base/java.lang.System.load(System.java:1837)
at com.apkspectrum.core.scanner.AaptNativeScanner.<clinit>(AaptNativeScanner.java:223)
at com.apkspectrum.core.scanner.AaptScanner.openApk(AaptScanner.java:51)
at com.apkspectrum.core.scanner.AaptLightScanner.openApk(AaptLightScanner.java:27)
at com.apkspectrum.core.scanner.ApkScanner.openApk(ApkScanner.java:105)
at com.apkscanner.gui.UiEventHandler$1.run(UiEventHandler.java:193)
at java.base/java.lang.Thread.run(Thread.java:829)
```

# Environment
1. OS: MacOS Ventura (13.0.1) M1
2. Java: openjdk11

**Comment by sunggyu-kam** [comment]
Hello.
To solve this problem, I have rebuilt the libaaptnativewrapper library.
But, I do not have a M1/M2 machine.
Can you test it by overwrite the attached library file?

[libAaptNativeWrapper.zip](https://github.com/apk-spectrum/apk-scanner/files/10234719/libAaptNativeWrapper.zip)

**Comment by d3xter-lab** [comment]
@sunggyu-kam Thanks for rebuilt. But when I run it after overwrite library, "error message' popup. It said '"libAaptNativeWrapper64.dylib" is damaged and can't be opened. You should move it to the Trash.' from macOS.

I confirmed that the binary is universal with arm64 included. However, it seems that something went wrong during the build process. Is there anything else I can check on my side?

**Comment by sunggyu-kam** [comment]
@d3xter-lab, Hello.
I don't know much about MAC.
And my build machine is mac pro 2013 which is very old.
Also, since June 21, Google is not supporting builds on Mac.
One guess is that changing the build issue file name could be the problem.
libAaptNativeWrapper.dylib -> libAaptNativeWrapper64.dylib

So I want to do some tests, but I can't confirm because I don't have m1.
Could you test the attached files as well?

testAaptNaiveWrapper.zip contains 3 folders and 2 files libc++.dylib and libc++64.dylib.

Copy the libc++.dylib and libc++64.dylib files in common for all three cases.
And overwrite the libAaptNativeWrapper64.dylib file in each folder to check if it is running.

[testAaptNaiveWrapper.zip](https://github.com/apk-spectrum/apk-scanner/files/10242851/testAaptNaiveWrapper.zip)

And tell me which version of APK Scanner you are testing.

**Comment by d3xter-lab** [comment]
@sunggyu-kam Thanks for quick action. I downloaded the latest code from the repo and built it, so it's maybe 2.10.1. And when I checked test libraries, it work well.

1. Fail, same error
2. Success
3. Success

But there was a signing problem. OS said that "this is a library built by unauthenticated developers". This problem was solved right away because it was an issue that could be allowed in OS settings. As a result of testing 1 apk file, it seems to be normal. Which library should we use between 2 and 3?

**Comment by sunggyu-kam** [comment]
Hello. Thank you for good news.

If case 3 works normal and you can build your own apk scanner,
I'm sorry, but please test one more thing.

1. Remove '64' from the library file name in the '[libAaptNativeWrapper.zip](https://github.com/apk-spectrum/apk-scanner/files/10234719/libAaptNativeWrapper.zip)' file that was previously delivered and added to the core/tool/darwin folder.

2. Fix the 'core/src/main/java/com/apkspectrum/resource/_Rfile.java'
Remove '64' from BIN_AAPT_LIB64_MAC and BIN_AAPT_LIBC64_MAC.

BIN_AAPT_LIB64_MAC			(Type.BIN, "darwin/libAaptNativeWrapper.dylib", "mac"),
BIN_AAPT_LIBC64_MAC			(Type.BIN, "darwin/libc++.dylib", "mac"),

3. Check the operation after building

**Comment by d3xter-lab** [comment]
@sunggyu-kam I tried removing '64' from the previously shared files, but the error message is the same. It was not available through that library.

**Comment by d3xter-lab** [comment]
Plus, adb binary in the darwin folder also only supports intel architecture. In my case, I replaced it with the binary I use, but it seems that this one also needs to be replaced when updating later.

**Comment by sunggyu-kam** [comment]
> Plus, adb binary in the darwin folder also only supports intel architecture. In my case, I replaced it with the binary I use, but it seems that this one also needs to be replaced when updating later.

ADB is applied to the dev branch. I will be included in the next release.
https://github.com/apk-spectrum/apk-spectrum-core/commit/1a8d060afa702138d1b45e802cdc7260c7355cbf

**Comment by sunggyu-kam** [comment]
> @sunggyu-kam I tried removing '64' from the previously shared files, but the error message is the same. It was not available through that library.

I'm sorry, but please rebuild with the attached file and test it.
[rebuildAaptNativeWrapper.zip](https://github.com/apk-spectrum/apk-scanner/files/10244425/rebuildAaptNativeWrapper.zip)

**Comment by d3xter-lab** [comment]
Same. 'Damage' related error messages occur.

**Comment by sunggyu-kam** [comment]
@d3xter-lab, Hello.
It's been a lot of time to synchronize other OS and library code.
Lastly, can you test it on Mac M1?

I don't know why it doesn't work as it is, but
Perhaps it may be a signature problem.
I don't have an Apple development account, so it's hard to issue a signed version for the time being.

Please test in the following way.
ApkScanner.jar, libc++.dylib file is used in common and existing libc++64.dylib file is deleted.
Is it possible to check the operation by changing only the libAaptNativeWrapper64.dylib file of each folder?
It was change only the Install_name_tool options.
I'll commit it to a version that works out of the four.

Attachments : [new_test.tar.gz](https://github.com/apk-spectrum/apk-scanner/files/10334017/new_test.tar.gz)

**Comment by d3xter-lab** [comment]
@sunggyu-kam Hi, I just tested it.
All 4 libraries are working fine. But as you said, it's not signed by a trusted developer, so we have to manually allow the library in os settings. Except that, all good.

I don't know what the 4 builds are different, but I think you can choose anyone.
Thanks!!

**Comment by sunggyu-kam** [comment]
This issue was released to version 2.11.

https://github.com/apk-spectrum/apk-scanner/releases/tag/v2.11_r1
